src/build/GhosttyResources.zig

```diff
@@ -224,16 +224,76 @@ pub fn init(b: *std.Build, cfg: *const Config) !GhosttyResources {
         // https://developer.gnome.org/documentation/guidelines/maintainer/integrating.html
 
         // Desktop file so that we have an icon and other metadata
-        try steps.append(&b.addInstallFile(
-            b.path("dist/linux/app.desktop"),
-            "share/applications/com.mitchellh.ghostty.desktop",
-        ).step);
+        try steps.append(
+            formatService(
+                b,
+                cfg,
+                b.path("dist/linux/app.desktop"),
+                b.fmt(
+                    "share/applications/com.mitchellh.ghostty{s}.desktop",
+                    .{
+                        switch (cfg.optimize) {
+                            .Debug, .ReleaseSafe => "-debug",
+                            .ReleaseFast, .ReleaseSmall => "",
+                        },
+                    },
+                ),
+            ),
+        );
+        // Service for DBus activation.
+        try steps.append(
+            formatService(
+                b,
+                cfg,
+                b.path("dist/linux/dbus.service"),
+                b.fmt(
+                    "share/dbus-1/services/com.mitchellh.ghostty{s}.service",
+                    .{
+                        switch (cfg.optimize) {
+                            .Debug, .ReleaseSafe => "-debug",
+                            .ReleaseFast, .ReleaseSmall => "",
+                        },
+                    },
+                ),
+            ),
+        );
+        // systemd user service
+        if (!cfg.flatpak)
```

Comment:
It might be better to add new systemd configuration option and use that to control this (and the associated dbus service patching). That should allow non-systemd distros like Alpine to package Ghostty without patching as well.

Reply:
I sort of had that in the initial PR. Would Alpine even support DBus activation, which is the only way the systemd service would get started? I've never used it outside of a container so I don't know what sort of environment we're dealing with there.

Reply:
You can run a desktop environment, which would include dbus-daemon/dbus-broker, both of which supports D-Bus activation.

An another distro of interest is Gentoo, where the no-systemd option is still supported.